client/components/activity-card/toolbar/actions-button.tsx

```diff
@@ -0,0 +1,164 @@
+/**
+ * External dependencies
+ */
+import { useTranslate } from 'i18n-calypso';
+import React, { useRef, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getActionableRewindId } from 'calypso/lib/jetpack/actionable-rewind-id';
+import { settingsPath } from 'calypso/lib/jetpack/paths';
+import { backupDownloadPath, backupRestorePath } from 'calypso/my-sites/backup/paths';
+import getDoesRewindNeedCredentials from 'calypso/state/selectors/get-does-rewind-need-credentials';
+import { recordTracksEvent } from 'calypso/state/analytics/actions/record';
+import { getSiteSlug, isJetpackSiteMultiSite } from 'calypso/state/sites/selectors';
+import ExternalLink from 'calypso/components/external-link';
+import Gridicon from 'calypso/components/gridicon';
+import PopoverMenu from 'calypso/components/popover/menu';
+import Button from 'calypso/components/forms/form-button';
+import { Activity } from '../types';
+
+/**
+ * Style dependencies
+ */
+import missingCredentialsIcon from 'calypso/components/jetpack/daily-backup-status/missing-credentials.svg';
+import downloadIcon from './download-icon.svg';
+
+type SingleSiteOwnProps = {
+	siteId: number;
+	siteSlug: string;
+	rewindId: string;
+};
+
+const SingleSiteActionsButton: React.FC< SingleSiteOwnProps > = ( {
+	siteId,
+	siteSlug,
+	rewindId,
+} ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const buttonRef = useRef( null );
+	const [ isPopoverVisible, setPopoverVisible ] = useState( false );
+	const togglePopoverMenu = () => {
```

Comment:
Using the `useCallback` hook could prevent unnecessary renders of the child components. Definitely not a blocker.

Reply:
I gave myself a challenge to make this component as efficient as possible, so I'm glad you pointed this out @monsieur-z! I debated using `useCallback` here, but my hunch was that it wouldn't necessarily save us much in the way of re-renders. This is what went through my mind, let me know if I'm missing something -- I'd love to improve on what I've got!

* At the component level, all three props are constant per Activity card AFAIK, so I wouldn't expect a re-render to happen as a result of a props change.
* At the level of the `togglePopoverMenu` function itself, `dispatch` and `recordTracksEvent` are both guaranteed to be constant, leaving `isPopoverVisible` and `setPopoverVisible` as the only two changing dependencies. We need to know the value of `isPopoverVisible` to toggle it, and it changes value by definition every time the function is called, so my thinking is that the function would be re-created on every render no matter what.

Reply:
You're absolutely right! I didn't check how `SingleSiteActionsButton` was instanciated.